base_contact_function_partner_firstname/res_partner.py

```diff
@@ -57,9 +57,6 @@ def create(self, cr, user, vals, context=None):
             vals['firstname'] = contact_info.firstname
             vals['title'] = contact_info.title.id or ''
 
-        if ('contact_id'in vals and ('reset_password' in context or
-                                     'install_mode' in context)):
-            return vals['contact_id']
```

Comment:
What does this statement do anyways? It looks like it prevents creation to be done are returns an existing object.
I think you said `"install_mode"` is in context while importing with the import button in the web interface.